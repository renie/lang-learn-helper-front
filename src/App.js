import React, { useEffect } from 'react'
import { getAllWords } from './actions';
import { 
  useSelector, 
  useDispatch } from 'react-redux'

import './App.css'
import 'bulma/css/bulma.min.css'

import Header from './layout/Header'
import WordList from './word/WordList'


const App = ({
  useEffectFn = useEffect,
  useSelectorFn = useSelector,
  useDispatchFn = useDispatch}) => {

  const words = useSelectorFn(store => store.wordState.words)
  const dispatch = useDispatchFn()
  const getAllWordsToDispatch = () => getAllWords()(dispatch)

  useEffectFn(()=> getAllWordsToDispatch(), [])

  return (
    <main className="app container">
      <Header />
      <WordList words={words}/>
    </main>
  )
} 

export default App
