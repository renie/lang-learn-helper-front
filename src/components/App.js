import React, { useEffect } from 'react'
import { useSelector, useDispatch } from 'react-redux'

import './App.css'
import 'bulma/css/bulma.min.css'

import { getAllWords } from '../actions'
import Header from './layout/Header'
import { WordListComponent } from './word'


export const App = ({ useEffect, useSelector, useDispatch, getAllWords}) => {
    const words = useSelector(store => store.wordState.words)
    const dispatch = useDispatch()
    const getAllWordsToDispatch = () => getAllWords()(dispatch)

    useEffect(()=> {getAllWordsToDispatch()})

    return (
        <main className="app container">
            <Header />
            <WordListComponent words={words}/>
        </main>
    )
} 


export const AppComponent = () => 
    App({useEffect, useSelector, useDispatch, getAllWords})


