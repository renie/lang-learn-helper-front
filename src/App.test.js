import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'

it('renders without crashing', () => {
    const useSelector = fn => fn({wordState: { words: [] }})
    const useDispatch = () => ({})

    const div = document.createElement('div')
    ReactDOM.render(<App useSelectorFn={useSelector} useDispatchFn={useDispatch}/>, div)
    ReactDOM.unmountComponentAtNode(div)
});
