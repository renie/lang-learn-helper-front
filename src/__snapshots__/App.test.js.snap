// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main
  className="app container"
>
  <header>
    <h1
      className="title"
    >
      Lang Helper
    </h1>
  </header>
  <div
    className="essayEditor"
    contentEditable="true"
    onInput={[Function]}
  />
  <ul
    className="mansory"
  >
    <li>
      No words do be listed
    </li>
  </ul>
</main>
`;
